src/OpenTelemetry.Api/Trace/TelemetrySpanNew.cs

```diff
@@ -0,0 +1,187 @@
+    /// </summary>
+    public class TelemetrySpanNew : IDisposable
+    {
+        internal static readonly TelemetrySpanNew NoopInstance = new TelemetrySpanNew(null);
+        private readonly Activity activity;
+
+        internal TelemetrySpanNew(Activity activity)
+        {
+            this.activity = activity;
+        }
+
+        /// <summary>
+        /// Gets the span context.
+        /// </summary>
+        public SpanContextNew Context
+        {
+            get
+            {
+                if (this.activity == null)
+                {
+                    return default;
+                }
+                else
+                {
+                    return new SpanContextNew(this.activity.Context);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this span will be recorded.
+        /// </summary>
+        public bool IsRecording
+        {
+            get
+            {
+                return (this.activity == null) ? false : this.activity.IsAllDataRequested;
+            }
+        }
+
+        /// <summary>
+        /// Sets the status of the span execution.
+        /// </summary>
+        public Status Status
+        {
+            set
+            {
+                this.activity?.SetStatus(value);
+            }
+        }
+
+        /// <summary>
+        /// Updates the <see cref="TelemetrySpanNew"/> name.
+        ///
+        /// If used, this will override the name provided via StartSpan method overload.
+        /// Upon this update, any sampling behavior based on <see cref="TelemetrySpanNew"/> name will depend on the
+        /// implementation.
+        /// </summary>
+        /// <param name="name">Name of the span.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void UpdateName(string name)
+        {
+            if (this.activity != null)
+            {
+                this.activity.DisplayName = name;
+            }
+        }
+
+        /// <summary>
+        /// Sets a new attribute on the span.
+        /// </summary>
+        /// <param name="key">Attribute key.</param>
+        /// <param name="value">Attribute value. The value may be an <see cref="IEnumerable"/> of primitive types. An enumeration may be iterated multiple times.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void SetAttribute(string key, string value)
+        {
+            this.activity?.AddTag(key, value);
+        }
+
+        /// <summary>
+        /// Adds a single <see cref="EventNew"/> to the <see cref="TelemetrySpanNew"/>.
+        /// </summary>
+        /// <param name="name">Name of the <see cref="Event"/>.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void AddEvent(string name)
+        {
+            this.activity?.AddEvent(new EventNew(name).ActivityEvent);
```

Comment:
@cijothomas Sorry I didn't notice this before. In these `AddEvent` calls, why bother doing the `new EventNew`? Seems like we could just new a `new ActivityEvent` directly. Maybe we don't even need `EventNew` if this is all it is used for?

Reply:
You are right. Since we dont expose AddEvent(Event event), there is no need of EventNew , atleast for now.